### PR TITLE
overload operators for solutions

### PR DESCRIPTION
**Pull request by mueldgog**
This change allows us to create, print, and access solutions more nicely:
```
x = m.addVar(lb=0,ub=4)
s = m.createSol()
s[x] = 2.0
print(s)
print(s[x])
```

### CODE REVIEW

**Comment by mueldgog** [comment]
The change in `__repr__` is a bit more lengthy because I didn't want to use `m.getVars()` because this would create a lot of unneeded python objects.

**Comment by mueldgog** [comment]
We could also drop `getSolVal()` and `setSolVal()` if we want. They are kind of redundant now.

File: src/pyscipopt/scip.pyx [hunk 1]
```diff
@@ -474,15 +474,38 @@ cdef class NLRow:
...
+        PY_SCIP_CALL(SCIPsetSolVal(self.scip, self.sol, var.scip_var, value))
+
+    def __repr__(self):
+        cdef SCIP_VAR* scip_var
+
+        vals = {}
+        for i in range(SCIPgetNVars(self.scip)):
+            scip_var = SCIPgetVars(self.scip)[i]
+
+            # exctract name
```
Comment (mattmilten) [inline src/pyscipopt/scip.pyx]:
```suggestion
            # extract name
```

File: src/pyscipopt/scip.pyx [hunk 2]
```diff
@@ -474,15 +474,38 @@ cdef class NLRow:
...
+            print(scip_var == NULL)
```
Comment (mattmilten) [inline src/pyscipopt/scip.pyx]: Do we need this `print`?

**Comment by mattmilten** [comment]
Cool! That makes working with the solution object much more convenient.

File: src/pyscipopt/scip.pyx (hunk 2 above)
Comment (mueldgog) [reply src/pyscipopt/scip.pyx]: Oh, sorry. I was just debugging a bit and accidentally added it.

Comment (mueldgog) [reply src/pyscipopt/scip.pyx]: Fixed with ee2df06.

**Pull request merged**